Added installation page (#844)
thefrontside/effection: www/docs/installation.mdx

Effection can be used in any JavaScript environment including Deno, Node and browsers. Effection packages are side-effect free and do not pollute your environment in any way, making it compatible with most JavaScript runtimes.
If you encounter obstacles integrating with your environment, please create a [GitHub Issue in the Effection repository](https://github.com/thefrontside/effection/issues/new).

## Node.js & Browser

<p class="inline-flex space-x-2 my-0">
  <img class="my-0" src="https://badgen.net/bundlephobia/min/effection" alt="Bundlephobia badge showing effection minified" />
  <img class="my-0" src="https://badgen.net/bundlephobia/minzip/effection" alt="Bundlephobia badge showing effection gzipped size" />
  <img class="my-0" src="https://badgen.net/bundlephobia/tree-shaking/effection" alt="Bundlephobia badge showing it's treeshackable" />
</p>

> Effection packages are available via [NPM][npm], [Yarn][yarn], [UNPKG][unpkg]. The package includes ESM, CommonJS and TypeScript types.

Reviewer: ```suggestion
Effection is available on [NPM][npm], as well as derived registries such as [Yarn][yarn] and [UNPKG][unpkg]. It comes with TypeScript types and can be consumed as both ESM and CommonJS.
```

```bash
// install with npm
npm install effection@next

// install with yarn
yarn add effection@next
```

## Deno

> Effection has first-class Deno support because it's written in Deno. We publish NPM/Yarn packages from our [Deno package](https://deno.land/x/effection). For example, to import the main function:

Reviewer: ```suggestion
Effection has first class support for Deno because it is developed with [Deno](https://deno.land). Releases are published to [https://deno.land/x/effection](https://deno.land/x/effection). For example, to import the `main()` function:
```
